.github/workflows/autobuild.yaml

```diff
@@ -1,41 +1,124 @@
-    - name: Mark as draft
+        sudo apt-get install jq
+    - name: Clone scripts
+      uses: actions/checkout@v2
+    - name: Pull repo from the inputs
+      if: ${{ github.event.inputs.repo == '' }}
+      run: |
+        echo "LLVM_REPO=llvm/llvm-project" >> commit.env
+    - name: Use the default repo
+      if: ${{ github.event.inputs.repo != '' }}
+      run: |
+        echo "LLVM_REPO=${{ github.event.inputs.repo }}" >> commit.env
+    # Choose the commit to build a release from.
+    #
+    # We want to avoid unbuildable revisions: choose the last green from CI.
+    # FIXME: The criteria should be some consistent set of buildbots passing.
+    # Use clangd/actions/pick after
+    # https://github.com/ramasilveyra/last-successful-gh-commit/issues/2 has
+    # been addressed.
+    - name: Get commit hash for LLVM head
+      if: ${{ github.event.inputs.commit == '' }}
       run: >
-        curl --fail --show-error -XPATCH
-        "-HAuthorization: Bearer ${{ secrets.RELEASE_TOKEN }}"
-        "https://api.github.com/repos/${{ github.repository }}/releases/${{ github.event.release.id }}"
-        "-d" '{"draft": true}'
-    - name: Persist release info
-      uses: actions/upload-artifact@v1
+        COMMIT=$(curl --fail --show-error
+        "https://api.github.com/repos/llvm/llvm-project/commits/main" |
+        jq ".sha" -r)
+
+        echo "LLVM_COMMIT=$COMMIT" >> commit.env
+    - name: Push picked commit to the file
+      if: ${{ github.event.inputs.commit != '' }}
+      run: |
+        echo "LLVM_COMMIT=${{ github.event.inputs.commit }}" >> commit.env
+    - name: Upload result
+      uses: actions/upload-artifact@v2
       with:
-        name: release
+        name: env
         path: commit.env
+  create_release:
+    name: Create release
+    runs-on: ubuntu-latest
+    needs: pick
+    steps:
+    - name: Fetch target commit
+      uses: actions/download-artifact@v1
+      with:
+        name:
+          env
+    - name: Set target commit
+      run: |
+        cat env/commit.env >> $GITHUB_ENV
+    - name: Compute release info
+      run: |
+        echo "RELEASE_COMMIT_SHORT=$(printf '%.12s' ${{ env.LLVM_COMMIT }})" >> $GITHUB_ENV
+        echo "RELEASE_DATE=$(date -u +%Y%m%d)" >> $GITHUB_ENV
+    - name: Pull tag name
+      if: ${{ github.event.inputs.tag != '' }}
+      run: >
+        echo "TAG_NAME=${{ github.event.inputs.tag }}" >> $GITHUB_ENV
+    - name: Automatically create tag name
+      if: ${{ github.event.inputs.tag == '' }}
+      run: >
+        echo "TAG_NAME=snapshot_${{ env.RELEASE_DATE }}" >> $GITHUB_ENV
+    - name: Create release
+      uses: actions/create-release@master
+      id: create_release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ env.TAG_NAME }}
+        release_name: ${{ env.RELEASE_DATE }} @${{ env.RELEASE_COMMIT_SHORT }}
+        body: |
+          Unstable snapshot of clangd on ${{ env.RELEASE_DATE }}.
```

Comment:
this line is only appropriate for the snapshots, and should probably be a workflow_dispatch input. (The line below is good though)